src/base/worker_pool.cc

```diff
@@ -36,11 +38,27 @@ void WorkerPool::AddWorker(Literal name, const NetWorker& worker, ui32 count) {
 void WorkerPool::AddWorker(Literal name, const SimpleWorker& worker,
                            ui32 count) {
   CHECK(count);
-  auto closure = [this, worker] { worker(is_shutting_down_); };
+  auto closure = [this, worker] { worker(this); };
   for (ui32 i = 0; i < count; ++i) {
     workers_.emplace_back(name, closure);
   }
 }
 
+bool WorkerPool::WaitUntilShutdown(
+    const std::chrono::duration<double, std::milli>& duration) {
+  if (is_shutting_down_) {
+    return true;
+  }
+
+  if (duration == std::chrono::duration<double, std::milli>::zero()) {
```

Comment:
This value should be private `static const` member for class - since it's used in more than one place.